dimagi/commcare-hq
# Bulk User Import Custom User Data Support targeted Required Field

## Product Description

This adds bulk user import validation for the backend changes done in [this PR](https://github.com/dimagi/commcare-hq/pull/35218). Previously, the "required_for" configurations did not apply during bulk user import.

## Technical Summary

[USH-5013
](https://dimagi.atlassian.net/browse/USH-5013)

## Feature Flag

no FF

## Safety Assurance

### Safety story

Tested locally and on staging. The change adds additional validation and impacts only bulk user import.

### Automated test coverage


I added test that covered validation of Custom User Data Fields during bulk user import.

### QA Plan


QA did a sanity check of this feature in this ticket [QA-7180](https://dimagi.atlassian.net/browse/QA-7180?atlOrigin=eyJpIjoiMjNmMGM5NGJiNzljNDEyNGJiODc4ODdjMDE5ODg1ODUiLCJwIjoiaiJ9)

### Rollback instructions




### Labels & Review


[QA-7180]: https://dimagi.atlassian.net/browse/QA-7180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
